File: RetrofitSample/app/src/main/java/com/xyz/networkcalls/Network.java
package com.xyz.networkcalls;

import okhttp3.OkHttpClient;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class Network {

    private static final String BASE_URL = "https://jsonplaceholder.typicode.com/";

    /*
    This method gives the retrofit instance
     */
    public static retrofit2.Retrofit getRetrofitInstance() {
        Retrofit.Builder builder = new Retrofit.Builder().baseUrl(BASE_URL)
                //In most cases requests to a server, and the responses from the server, are not Java objects. They’re mapped to some language neutral format like JSON
                .addConverterFactory(GsonConverterFactory.create())
                //this client will help you to print the API call logs in the Logcat
                .client(new OkHttpClient.Builder().build());
        return builder.build();
    }
}

File: evaluations/evaluation_week16/app/src/main/java/com/xyz/evaluations_week16/Network.kt
package com.xyz.evaluations_week16

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class Network {

    /**
     * Companion object is like static object in java
     */
    companion object {

        private val httpLoggingInterceptor: HttpLoggingInterceptor =
            HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY)

        /**
         * This method will give the instance of retrofit and it can be directly accessed
         * from MainActivity like Network.getInstance()
         */
        fun getInstance(): Retrofit {
            return Retrofit.Builder()
                .baseUrl("https://reqres.in/")
                .addConverterFactory(GsonConverterFactory.create())
                .client(OkHttpClient.Builder().addInterceptor(httpLoggingInterceptor).build())
                .build()
        }
    }
}